src/NuGet.Core/NuGet.CommandLine.XPlat/NuGet.CommandLine.XPlat.csproj

```diff
@@ -20,23 +20,13 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.CommandLineUtils.Sources" PrivateAssets="All" />
     <PackageReference Include="System.Diagnostics.Debug" />
+    <PackageReference Include="Microsoft.Build" ExcludeAssets="runtime" />
   </ItemGroup>
 
-  <!-- Microsoft.Build.Locator is needed when debugging, but should not be used in the assemblies we insert.
-       It also generates a build error if the output directory contains MSBuild dlls, so exclude msbuild assets when using the locator. -->
-  <Choose>
-    <When Condition=" '$(Configuration)' == 'Debug' ">
-      <ItemGroup>
-        <PackageReference Include="Microsoft.Build" ExcludeAssets="runtime" />
-        <PackageReference Include="Microsoft.Build.Locator" />
-      </ItemGroup>
-    </When>
-    <Otherwise>
-      <ItemGroup>
-        <PackageReference Include="Microsoft.Build" />
-      </ItemGroup>
-    </Otherwise>
-  </Choose>
+  <!-- Microsoft.Build.Locator is only used when debugging, and the compiler will trim this dependency from Release assemblies we insert. -->
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Build.Locator" PrivateAssets="All" />
```

Comment:
Why remove the Choose/When instead that ensured we don't accidentally use this dependency during our release builds?

Reply:
I feel like this `<Choose>` is misleading.  During a local restore in visual studio or command-line, the Configuration property is almost always Debug.  In CI, we set it as a global property which runs restore and then later on the build uses what comes from restore.  Don't we tell end users that the only "supported" condition for package references is `TargetFramework`?

What really pulls in MSBuildLocator is this:

https://github.com/NuGet/NuGet.Client/blob/d43053929551ef55dbb524dacba0bcc37079b5dd/src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs#L39-L60

In my opinion a compilation condition is better since all of that logic is in one place.

Reply:
We've offered advice that you can use Choose/When with the obvious stomping behavior when you do so. 

I don't feel too strongly about. I just wanted to make sure we understand the consequences of that change.

Reply:
> Why remove the Choose/When instead that ensured we don't accidentally use this dependency during our release builds?

I wrote the following comment after testing with Jeff and being convinced that the Condition didn't really have any effect on whether the assembly was pulled in or not. In other words, this is removing dead code, and the comment attempts to remind the reader about the compiler's trimming action:
`Microsoft.Build.Locator is only used when debugging, and the compiler will trim this dependency from Release assemblies we insert`

Reply:
what do you mean "whether the assembly was pulled in or not"?

By not conditioning the package, it will become a dependency of `NuGet.CommandLine.XPlat.{version}.nupkg`, and therefore when the .NET SDK out package, it will transitively bring in the MSBuild locator.  That will bloat (slightly) the .NET SDK download & install size for all customers, including CI builds (including our own) that use `dotnet-install.{ps1|sh}`. The .NET SDK doesn't need the locator to locate itself.

> the compiler will trim this dependency from Release assemblies we insert

Just because the compiler doesn't add a compiled in reference to the assembly doesn't mean that the package also gets trimmed as a dependency.

Reply:
Hope we can all get on the same page in next week's sync. 
In the meantime, I took a screenshot showing the diff of the nupkg references before/after this change.
Note that `Microsoft.Build.Locator` is not referenced in either of the Release builds of this package, neither before this change, nor after this change.

**Left**: Before this PR references in ([NuGet.CommandLine.XPlat.6.3.0.nupkg](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6587874&view=results))
**Right**: This PR - nupkg references ([NuGet.CommandLine.XPlat.6.4.0-xprivate.nupkg](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6613657&view=results))

![image](https://user-images.githubusercontent.com/49205731/186999124-60c08f5d-6db6-470f-8358-e66d3b4e6691.png)

Reply:
Oops, yes the `PrivateAssets="all"` on the PackageReference prevents it becoming a dependency.

Ok, so I agree that removing the condition on the package reference has no functional changes. It's just a question of whether we want compile-time enforcement that the MSBuild Locator can't be used in release builds vs how much it bothers people to condition the package on debug only. The risk is that if someone accidentally uses the locator in a release build, then the .NET SDK will fail at runtime. If we're lucky our tests will catch it. If we're unlucky, it could ship to customers.

Reply:
Perhaps it would be better if we keyed off a different property other than `Configuration` to determine if we want a compile-time dependency on this.  

```xml
<!-- Only reference Microsoft.Build.Locator for non-official builds because official builds place NuGet next to MSBuild so it doesn't need to be located at run time -->
<PackageReference Include="Microsoft.Build.Locator" Condition="'$(IsOfficialBuild)' != 'true'" />
```

For me that seems more explicit if we want to make sure that no official build ever takes a dependency on it.  We could even update the compile-time constant 
```c#
#if !OFFICIAL_BUILD
 // Use Microsoft.Build.Locator
#endif
```

Reply:
Offline discussion lead us to proposal # 3 below...

1. Reinstate the Choose with `$(Configuration)`
2. ~`Attempt to use a "IsCI" (IsTestBits) condition instead of Configuration~ (this could lead to confusing behavior)
3. Comment out the code utilizing MSBuildLocator. If you want to debug, uncomment it. Also Comment out PackageReference for MSBuildLocator. Update local dev docs if you need to debug `dotnet nuget`.